### PR TITLE
Issue numbering across streams

### PR DESCRIPTION
**Pull request by sharonwarner**
Connects #3390 

screen shot 2017-11-03 at 10 12 21 am

### CODE REVIEW

File: client/app/hearings/components/HearingWorksheetStream.jsx [hunk 1]
```diff
@@ -30,19 +31,28 @@ class HearingWorksheetStream extends Component {
...
+
     return <div className="cf-hearings-worksheet-data">
       <h2 className="cf-hearings-worksheet-header">Issues</h2>
       {Object.values(worksheetAppeals).map((appeal, key) => {
 
+        const appealWorksheetIssues = IssuesUtil.filterIssuesOnAppeal(worksheetIssues, appeal.id);
+        const currentIssueCount = issueCount;
+
+        issueCount += _.size(appealWorksheetIssues);
+
```
Comment (sharonwarner) [inline client/app/hearings/components/HearingWorksheetStream.jsx]: I hate this... there has to be a better way to do it, right?

File: client/app/hearings/util/IssuesUtil.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import _ from 'lodash';
+
+const IssuesUtil = {
+  filterIssuesOnAppeal(issues, appealId) {
+    return _.pickBy(issues, (issue) => {
```
Comment (nickheiner-usds) [inline client/app/hearings/util/IssuesUtil.js]: Lodash golf: 

```js
    return _(issues)
      .omitBy('_destroy')
      .pickBy({appeal_id: appealId})
      .value()
```

File: client/app/hearings/components/HearingWorksheetStream.jsx (hunk 1 above)
Comment (nickheiner-usds) [reply client/app/hearings/components/HearingWorksheetStream.jsx]: I agree that it doesn't feel great, but I think it's fine in this instance.

File: client/app/hearings/components/HearingWorksheetIssues.jsx [hunk 3]
```diff
@@ -128,5 +124,6 @@ export default connect(
 
 HearingWorksheetIssues.propTypes = {
   appealKey: PropTypes.number.isRequired,
-  worksheetStreamsAppeal: PropTypes.object.isRequired
+  worksheetStreamsAppeal: PropTypes.object.isRequired,
+  issueCount: PropTypes.number.isRequired
```
Comment (nickheiner-usds) [inline client/app/hearings/components/HearingWorksheetIssues.jsx]: This would read much better to me as `previousIssuesCount` or something like that.

Comment (nickheiner-usds) [reply client/app/hearings/components/HearingWorksheetIssues.jsx]: or `countOfIssuesInPreviousAppeals`.

File: client/app/hearings/util/IssuesUtil.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import _ from 'lodash';
+
+const IssuesUtil = {
```
Comment (nickheiner-usds) [inline client/app/hearings/util/IssuesUtil.js]: Let's chat in person about how to clean this up a bit.

**Comment by nickheiner-usds** [comment]
This will be good to go after the feature test is in.

File: spec/feature/hearings_spec.rb [hunk 5]
```diff
@@ -170,6 +170,22 @@
       expect(page).to_not have_content("Service Connection")
     end
 
+    context "Multiple appeal streams" do
+      before do
+        vbms_id = Hearing.find(1).appeal.vbms_id
+        Generators::Appeal.create(vbms_id: vbms_id, vacols_record: { template: :pending_hearing })
+      end
+
+      scenario "Numbering is consistent" do
```
Comment (nickheiner-usds) [inline spec/feature/hearings_spec.rb]: Just for fun, would you mind wrapping this in `ensure_stable` and pushing it to Travis so we can see that it passes reliably on CI? Then we'd remove `ensure_stable` before merging.

File: spec/feature/hearings_spec.rb [hunk 6]
```diff
@@ -170,6 +170,22 @@
       expect(page).to_not have_content("Service Connection")
     end
 
+    context "Multiple appeal streams" do
```
Comment (nickheiner-usds) [inline spec/feature/hearings_spec.rb]: 👍

**Comment by sharonwarner** [comment]
screen shot 2017-11-06 at 11 39 47 am

**Pull request merged**